cephbackupstorage/cephbackupstorage/cephagent.py

```
@@ -1072,6 +1149,529 @@ def cancel(self, req):
+            return jsonobject.dumps(rsp)
+
+        # Zip-Slip pre-check: verify no path escapes before sending to remote host.
+        try:
+            listing = shell.call("tar -tf %s" % linux.shellquote(cmd.upgradePackagePath))
+            for entry in listing.strip().splitlines():
+                entry = entry.strip()
+                if entry.startswith('/') or linux.contains_path_traversal(entry):
+                    rsp.success = False
+                    rsp.error = "path escape detected in upgrade package entry: %s" % entry
+                    return jsonobject.dumps(rsp)
+        except Exception as e:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)